apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/McsClaimTransformerTest.java

```diff
@@ -0,0 +1,484 @@
+              new DataTransformer.ErrorMessage("idrHdrToDateOfSvc", "invalid date"),
+              new DataTransformer.ErrorMessage(
+                  "diagCode-0-idrClmHdIcn", "invalid length: expected=[1,15] actual=18"),
+              new DataTransformer.ErrorMessage(
+                  "diagCode-0-idrDiagIcdType", "invalid length: expected=[1,1] actual=22"),
+              new DataTransformer.ErrorMessage(
+                  "diagCode-0-idrDiagCode", "invalid length: expected=[1,7] actual=10"),
+              new DataTransformer.ErrorMessage("detail-0-idrDtlFromDate", "invalid date"),
+              new DataTransformer.ErrorMessage("detail-0-idrDtlToDate", "invalid date"),
+              new DataTransformer.ErrorMessage(
+                  "detail-0-idrProcCode", "invalid length: expected=[1,5] actual=8"),
+              new DataTransformer.ErrorMessage(
+                  "detail-0-idrModOne", "invalid length: expected=[1,2] actual=5"),
+              new DataTransformer.ErrorMessage(
+                  "detail-0-idrModTwo", "invalid length: expected=[1,2] actual=5"),
+              new DataTransformer.ErrorMessage(
+                  "detail-0-idrModThree", "invalid length: expected=[1,2] actual=5"),
+              new DataTransformer.ErrorMessage(
+                  "detail-0-idrModFour", "invalid length: expected=[1,2] actual=5"),
+              new DataTransformer.ErrorMessage(
+                  "detail-0-idrDtlDiagIcdType", "invalid length: expected=[1,1] actual=39"),
+              new DataTransformer.ErrorMessage(
+                  "detail-0-idrDtlPrimaryDiagCode", "invalid length: expected=[1,7] actual=10"),
+              new DataTransformer.ErrorMessage(
+                  "detail-0-idrKPosLnameOrg", "invalid length: expected=[1,60] actual=63"),
+              new DataTransformer.ErrorMessage(
+                  "detail-0-idrKPosFname", "invalid length: expected=[1,35] actual=38"),
+              new DataTransformer.ErrorMessage(
+                  "detail-0-idrKPosMname", "invalid length: expected=[1,25] actual=28"),
+              new DataTransformer.ErrorMessage(
+                  "detail-0-idrKPosAddr1", "invalid length: expected=[1,55] actual=58"),
+              new DataTransformer.ErrorMessage(
+                  "detail-0-idrKPosAddr2_1st", "invalid length: expected=[1,30] actual=33"),
+              new DataTransformer.ErrorMessage(
+                  "detail-0-idrKPosAddr2_2nd", "invalid length: expected=[1,25] actual=28"),
+              new DataTransformer.ErrorMessage(
+                  "detail-0-idrKPosCity", "invalid length: expected=[1,30] actual=33"),
+              new DataTransformer.ErrorMessage(
+                  "detail-0-idrKPosState", "invalid length: expected=[1,2] actual=5"),
+              new DataTransformer.ErrorMessage(
+                  "detail-0-idrKPosZip", "invalid length: expected=[1,15] actual=18")),
+          ex.getErrors());
```

Comment:
Interesting question.  There will ultimately be many more fields and I'm not thrilled by having hundreds of separate unit tests.  The individual field validations are independent of one another.  I can improve how the individual errors are reported in the test so it shows just what's different rather than the whole lists being shown.

Reply:
My thinking on something like this is that you are going to have a hundred of something no matter what you do. A hundred one line tests is more lines of code but they are simpler to understand and work with. Another option is a parametrized test where you don't even need a separate method for each test but you get the same effect when you run and debug them. Just some thoughts.